src/Fixer/Phpdoc/PhpdocAddMissingParamAnnotationFixer.php

```diff
@@ -189,16 +189,16 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
             $newLines = [];
 
             foreach ($arguments as $argument) {
-                $type = $argument['type'] ?: 'mixed';
-
-                if (!str_starts_with($type, '?') && 'null' === strtolower($argument['default'])) {
-                    $type = 'null|'.$type;
+                if ('' === $argument['type']) {
+                    $argument['type'] = 'mixed';
+                } elseif ('mixed' !== $argument['type'] && ('?' === $argument['type'][0]  || 'null' === strtolower($argument['default']))) {
```

Comment:
`'?' === $argument['type'][0]`
->
`str_starts_with($type, '?')`

Reply:
@SpacePossum slower

Reply:
I prefer readable in both the code and the diff here

Reply:
@SpacePossum I'm understand you, what's reason to restore? old version worked slower

Reply:
This `str_starts_with($type, '?')` is more easy to read than `'?' === $argument['type'][0]` for example.
I'm not saying it is faster, I care more about readability here than about the speed

Reply:
@SpacePossum for end users, speed is more important, I don't think that such code is incomprehensible to senior  developers sending PR's

Reply:
I won't merge the changes "as is", so fine by me to leave like this in the PR but note that you'll have to find another member to help this get merged in that case